- Fixes issue#2413 - Tooltip - Need support to disable auto hiding  of tooltip when touching/scrolling anywhere inside the active tooltip popover container

anandhakumarpalanisamy opened the pull request:
**Fixes [issue#2413](https://github.com/react-native-elements/react-native-elements/issues/2413) - Tooltip - Need support to disable auto hiding of tooltip when touching/scrolling anywhere inside the active tooltip popover container**

- Added a configuration parameter **"closeOnlyOnBackdropPress"** to the tooltip

- **closeOnlyOnBackdropPress** -> default value is false -> default tooltip behaviour

- **closeOnlyOnBackdropPress** -> when set to true
    - Disables auto hiding of tooltip when touching/scrolling anywhere inside the active tooltip popover container
    - Tooltip closes only when overlay backdrop is pressed (or) highlighted tooltip button is pressed


 **Demo**

![react-native-elements-tooltip-fix](https://user-images.githubusercontent.com/30879156/89315419-17646b80-d67b-11ea-8ece-258d137a54d5.gif)

anandhakumarpalanisamy commented:
What is the estimated time duration for a pull request to be reviewed and merged ?

Thanks & regards,
Anand

flyingcircle commented:
@anandhakumarpalanisamy thank you for your submission. I've been working pretty hard on trying to get the number of tickets down since taking on this project. I'm slating this for a `2.3.0` release. I've just submitted a review on what I think would be a good change before merging. If possible, it would also be good to add some unit tests to cover the code that you're adding. Splitting that function into separate components should help with that piece as well. Thanks again!

anandhakumarpalanisamy commented:
> @anandhakumarpalanisamy thank you for your submission. I've been working pretty hard on trying to get the number of tickets down since taking on this project. I'm slating this for a `2.3.0` release. I've just submitted a review on what I think would be a good change before merging. If possible, it would also be good to add some unit tests to cover the code that you're adding. Splitting that function into separate components should help with that piece as well. Thanks again!

@flyingcircle  Thank you very much for your reply . Wish you all the best for your hard work on reducing the number of tickets.
I have added the unit tests and now all checks are passed.

**Unit Test Experience :** 

**Problem :** Jest tests were passing locally but was failing on Travis CI 
- It took me long time to identify the cause of the problem as i was not familiar of these workflows.

**Cause :**  The test snapshots generated locally conflicts with the Travis CI setup.

**Solution:**  - Modified travis CI's config file(_".travis.yml"_) to update snapshots while doing the jest coverage tests as shown below. Now  travisCI will update snapshots while running the tests.

![Screenshot from 2020-08-14 20-21-05](https://user-images.githubusercontent.com/30879156/90280757-c539fc00-de6b-11ea-998a-af95fc115dd4.png)

I have added the fix in this pull request itself. 
If you think, what i have faced is a potential issue and  the snapshot fix is valid, kindly create a new issue or merge along with this pull request. 

-thanks & regards,
Anand

flyingcircle commented:
@anandhakumarpalanisamy actually that change to travis CI is incorrect. The Travis CI should not be updating the snapshots as that would break the entire point of having the CI. Please undo that change and if your tests continue to fail we candiscuss what is the cause of that and what may be needed to fix it.

anandhakumarpalanisamy commented:
> @anandhakumarpalanisamy actually that change to travis CI is incorrect. The Travis CI should not be updating the snapshots as that would break the entire point of having the CI. Please undo that change and if your tests continue to fail we candiscuss what is the cause of that and what may be needed to fix it.

@flyingcircle : Thanks for the inputs. I have followed your instructions and removed the travis ci config changes.
I have also updated the jest tests and **achieved 100 % diff coverage.** 
Kindly check them . 

-regards,
 Anand

flyingcircle commented on website/docs/tooltip.md:
```diff
@@ -54,6 +54,7 @@ import Modal from 'modal-react-native-web';
 - [`toggleAction`](#toggleaction)
 - [`skipAndroidStatusBar`](#skipandroidstatusbar)
 - [`ModalComponent`](#modalcomponent)
+- [`closeOnlyOnBackdropPress`](#closeOnlyOnBackdropPress)
```
should be:
```- [`closeOnlyOnBackdropPress`](#closeonlyonbackdroppress)```

anandhakumarpalanisamy replied:
Done :+1:

flyingcircle commented:
@anandhakumarpalanisamy I tried out your changes and they look good! The only thing I saw in reviewing the code is that a line in the documentation was wrong. Fix that and then I'll merge.

flyingcircle commented on website/docs/tooltip.md:
````diff
@@ -252,3 +253,24 @@ useEffect(() => {
+
+- When `true`, Tooltip closes only when overlay backdrop is pressed (or) highlighted tooltip button is pressed.
+
+```js
````
You can remove this js example. I think it's already apparent how to use this prop.

anandhakumarpalanisamy replied:
Done :+1:

anandhakumarpalanisamy commented:
> @anandhakumarpalanisamy I tried out your changes and they look good! The only thing I saw in reviewing the code is that a line in the documentation was wrong. Fix that and then I'll merge.

@flyingcircle  : Sorry for the late reply . I was little bit busy with another task . I have updated your suggestions now. Kindly check if that is fine . Kindly let me know , if it would require anymore changes. 

Thanks & regards
-Anand

panda919 commented:
closeOnlyOnBackdropPress effect is not working on my side. 
Source
`<SafeAreaView style={styles.container}><Text>Open up App.js to start working on your app!</Text>
            <View style={styles.modalContainer}>
                <Tooltip
                    popover={<Text>Hello world</Text>}
                    closeOnlyOnBackdropPress={true}
                    skipAndroidStatusBar={true}
                    backgroundColor="white">
                    <Text style={styles.text}>Hello World</Text>
                </Tooltip>
            </View>
        </SafeAreaView>`
`container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
    },
    modalContainer: {
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
    },
    text: {
        color: 'black',
    },`
But I got this effect.
![image](https://user-images.githubusercontent.com/25161426/112718905-e9e17080-8f30-11eb-9f97-843e8bfb1210.png)

harisbaig100 commented:
@panda919 install verson 2.3.0, that fixd the issue for me
